SimCalorimetry/EcalSimProducers/interface/EcalDigiProducer_Ph2.h

```diff
@@ -0,0 +1,122 @@
+#ifndef SimCalorimetry_EcalSimProducers_EcalDigiProducer_Ph2_h
+#define SimCalorimetry_EcalSimProducers_EcalDigiProducer_Ph2_h
+
+#include "SimCalorimetry/EcalSimAlgos/interface/APDShape.h"
+#include "SimCalorimetry/EcalSimAlgos/interface/EBShape.h"
+#include "DataFormats/Math/interface/Error.h"
+#include "FWCore/Framework/interface/ProducesCollector.h"
+#include "SimGeneral/NoiseGenerators/interface/CorrelatedNoisifier.h"
+#include "SimCalorimetry/EcalSimAlgos/interface/EcalCorrelatedNoiseMatrix_Ph2.h"
+
+#include "SimCalorimetry/CaloSimAlgos/interface/CaloTDigitizer.h"
+#include "SimCalorimetry/EcalSimAlgos/interface/EcalTDigitizer_Ph2.h"
+#include "SimCalorimetry/EcalSimAlgos/interface/EcalDigitizerTraits_Ph2.h"
+#include "SimGeneral/MixingModule/interface/DigiAccumulatorMixMod.h"
+#include "SimCalorimetry/EcalSimAlgos/interface/EBHitResponse.h"
+#include <vector>
+
+typedef EcalTDigitizer_Ph2<EBDigitizerTraits_Ph2> EBDigitizer_Ph2;
+
+class APDSimParameters;
+class CaloHitResponse;
+class EcalSimParameterMap_Ph2;
+class EcalLiteDTUCoder;
+class EcalElectronicsSim_Ph2;
+class EcalBaseSignalGenerator;
+class CaloGeometry;
+class EBDigiCollectionPh2;
+class PileUpEventPrincipal;
+
+namespace edm {
+  class ConsumesCollector;
+  class ProducerBase;
+  class Event;
+  class EventSetup;
+  template <typename T>
+  class Handle;
+  class ParameterSet;
+  class StreamID;
+}  // namespace edm
+
+namespace CLHEP {
+  class HepRandomEngine;
+}
+
+class EcalDigiProducer_Ph2 : public DigiAccumulatorMixMod {
```

Comment:
Though a separate DigiProducer class does involve some duplication, I understand the desire to separate it from the Phase0/1 producer, since Ph2 doesn't have to handle EE or ES. Accordingly, I'm okay with this duplication, but will be making a number of comments in the cc file to improve the code quality (since it should now be treated as new code).